src/NHibernate/Linq/NhRelinqQueryParser.cs

```diff
@@ -53,24 +54,33 @@ static NhRelinqQueryParser()
 		/// </summary>
 		/// <param name="expression">The expression to transform.</param>
 		/// <returns>The transformed expression.</returns>
-		[Obsolete("Use overload with an additional sessionFactory parameter")]
+		[Obsolete("Use overload with PreTransformationParameters parameter")]
 		public static Expression PreTransform(Expression expression)
 		{
-			return PreTransform(expression, null);
+			// In order to keep the old behavior use a DML query mode to skip detecting variables,
+			// which will then generate parameters for each constant expression
+			return PreTransform(expression, new PreTransformationParameters(QueryMode.Delete, null)).Expression;
 		}
 
 		/// <summary>
 		/// Applies the minimal transformations required before parametrization,
 		/// expression key computing and parsing.
 		/// </summary>
 		/// <param name="expression">The expression to transform.</param>
-		/// <param name="sessionFactory">The session factory.</param>
-		/// <returns>The transformed expression.</returns>
-		public static Expression PreTransform(Expression expression, ISessionFactoryImplementor sessionFactory)
+		/// <param name="parameters">The parameters used in the transformation process.</param>
+		/// <returns><see cref="PreTransformationResult"/> that contains the transformed expression.</returns>
+		public static PreTransformationResult PreTransform(Expression expression, PreTransformationParameters parameters)
```

Comment:
Not a breaking change as the overload was added in #1868, which is not yet released.